python/glow/wgr/model_functions.py

```diff
@@ -471,117 +474,168 @@ def __assert_all_present(df: pd.DataFrame, col_name: Any, df_name: str) -> None:
+                f"Column {label} is not binary. It has {num_non_binary_vals} non-binary value(s).",
                 UserWarning)
 
 
 @typechecked
-def __num_non_binary_values(s: pd.Series) -> int:
-    """
-    Returns the number of values in a series that are neither 0, 1, nor missing.
+def _is_binary(df: pd.DataFrame) -> bool:
     """
+    Checks whether a Pandas DataFrame is all binary.
 
-    non_binary_vals = (~s.dropna().isin([0, 1])).sum()
-    return int(non_binary_vals)
+    Args:
+        df : Pandas DataFrame
+    """
+    return df.isin([0, 1, None]).all(axis=None).item()
 
 
 @typechecked
-def __check_binary(df: pd.DataFrame) -> None:
+def _fill_na_and_standardize(pdf: pd.DataFrame) -> pd.DataFrame:
     """
-    Warns if any column of a pandas DataFrame is not a binary value equal to 0 or 1.
+    Replaces the NaNs in a Pandas DataFrame with the mean of the column and
+    mean centers and scales the columns.
 
     Args:
         df : Pandas DataFrame
     """
-    for label in df:
-        num_non_binary_vals = __num_non_binary_values(df[label])
-        if num_non_binary_vals != 0:
-            warnings.warn(f"Column {label} is not binary ({num_non_binary_vals} non binary values)",
-                          UserWarning)
+    pdf_mean = pdf.mean()
+    _pdf = pdf.fillna(pdf_mean)
+    pdf_std = pdf.std()
+    zero_std_labels = _pdf.columns[pdf_std == 0]
+    if zero_std_labels.empty:
+        return (_pdf - pdf_mean) / pdf_std
+    else:
+        raise ValueError(f"""Column(s) {zero_std_labels.tolist()} have zero std. dev.!""")
 
 
 @typechecked
-def __check_binary_or_standardized(df: pd.DataFrame) -> None:
+def _prepare_labels_and_warn(label_df: pd.DataFrame, is_binary: bool,
+                             label_type: str) -> pd.DataFrame:
     """
-    Warns if columns of a pandas DataFrame are neither binary nor standardized.
+    Gets a Pandas DataFrame containing phenotypes (label_df) and whether it is binary or not
+    and generates the appropriate warning based on label_type. Returns standardized label DataFrame
 
     Args:
-        df : Pandas DataFrame
+        label_df: Pandas DataFrame of phenotypes
+        is_binary: Whether label_df is binary or not
+        label_type: User's choice for label type treatment ('detect', 'binary', or 'quantitative')
+
+    Returns:
+        Standardized label Pandas Dataframe
     """
-    for label in df:
-        mean = df[label].mean()
-        std_dev = df[label].std()
-        num_non_binary_values = __num_non_binary_values(df[label])
-        continuous_ok = __is_zero(mean) and __is_one(std_dev)
-
-        if num_non_binary_values == 0:
-            # Valid binary trait
-            continue
-
-        # Check as continuous trait
-        __assert_all_present(df, label, 'label')
-        if not continuous_ok:
+    if label_type == 'detect':
+        if is_binary:
+            warnings.warn(
```

Comment:
Should this really be a warning? This is expected. Seems like we should log or print.